tools/travis/static-analysis

```diff
@@ -1,23 +0,0 @@
-#!/bin/bash
-
-set -e
-set -x
-
-retcode=0
-./tools/run-mypy --linecoverage-report || retcode=1
```

Comment:
Were we using the `--linecoverage-report`? It gets dropped in this move.

Reply:
Oh, I hadn't noticed that.  Well, the answer is that we're not using it anymore, since we just require 100% coverage these days.  So it's for the best, though should be its own commit.

Reply:
Especially because there's other code to delete from `run-mypy`.  

Though there is some value to keeping the code around, since I've had a few folks ask me how one tracked progress on mypy coverage and being able to point to it has been helpful.  But  probably we should just turn that thought into an issue in python/mypy and they can add the brief bit of docs on how to do this :)